assets/js/common/ApiKeySettingsModal/ApiKeySettingsModal.jsx

```diff
@@ -0,0 +1,191 @@
+    >
+      <div className="flex flex-col my-5">
+        <span className="my-2 mb-4 text-gray-500">
+          {' '}
+          By generating a new key, you will need to replace the API key on all
+          hosts.{' '}
+        </span>
+
+        <div className="flex space-x-1">
+          <div className="w-1/5">
+            <Label>Never Expires</Label>
+          </div>
+
+          <Switch
+            selected={apiKeyNeverExpires}
+            onChange={() => {
+              setapiKeyNeverExpires((enabled) => !enabled);
+              setQuantityError(false);
+            }}
+          />
+        </div>
+        <div className="flex items-center my-2 space-x-2">
+          <div className="w-1/3">
+            <Label>Key Expiration</Label>
+          </div>
+
+          <div className="w-2/4 pt-1">
+            <InputNumber
+              value={timeQuantity}
+              className="!h-8"
+              type="number"
+              min="0"
+              disabled={apiKeyNeverExpires}
+              error={quantityError}
+              onChange={(value) => {
+                setTimeQuantity(parseInt(value, 10));
+                setQuantityError(false);
+              }}
+            />
+          </div>
+          <div className="w-2/4 pt-4">
+            <Select
+              optionsName=""
+              options={timeOptions}
+              disabled={apiKeyNeverExpires}
+              value={timeQuantityType}
+              onChange={(value) => setTimeQuantityType(value)}
+            />
+          </div>
+          <div className="w-1/6 h-4/5">
+            <Button
+              onClick={() => generateApiKeyExpiration()}
+              disabled={quantityError || loading}
+            >
+              Generate
+            </Button>
+          </div>
+        </div>
+        {quantityError && (
+          <span className="my-1 mb-4 text-red-500">
+            {' '}
+            Key expiration value needs to be greater than 0{' '}
+          </span>
+        )}
+        {generatedApiKey && keyGenerated && !loading && (
+          <div className="flex flex-col my-1 mb-4">
+            <div className="flex space-x-2">
+              <div className="w-full break-words p-2 pr-2 rounded-lg bg-white border-gray-300 border">
+                <code> {truncate(generatedApiKey, { length: 65 })} </code>
+              </div>
+              <button
+                type="button"
+                aria-label="copy api key"
+                onClick={() => {
+                  window.navigator.clipboard.writeText(generatedApiKey);
+                }}
+              >
+                <EOS_CONTENT_COPY role="button" size="25" />
+              </button>
+            </div>
+            <div className="flex space-x-2">
+              <EOS_INFO_OUTLINED size="20" className="mt-2" />
+
+              <div className="mt-2 text-gray-600 text-sm">
+                {generatedApiKeyExpiration
+                  ? `Key Will Expire ${format(
```

Comment:
`Key will expire...` all words small case except the first one

Reply:
done, I was copying right from the mockups xD